pyro/infer/reparam/conjugate.py

```diff
@@ -88,3 +88,34 @@ def __call__(self, name, fn, obs):
         # Return an importance-weighted point estimate.
         new_fn = dist.Delta(value, log_density=log_density, event_dim=fn.event_dim)
         return new_fn, value
+
+
+class ConjugateLatentReparam(Reparam):
+    """
+    Reparameterize to add an extra ``pyro.deterministic`` site for the latent
+    variable of a conjugate pair distribution.
+
+    This only works for a small set of conjugate pair distributions, including
+    :class:`~pyro.distributions.BetaBinomial` ,
+    :class:`~pyro.distributions.DirichletMultinomial` ,
+    :class:`~pyro.distributions.GammaPoisson` , and
+    :class:`~pyro.distributions.GaussianHMM` .
+    """
+    def __call__(self, name, fn, obs):
+        assert obs is None, "ConjugateLatentReparam does not support observe statements"
```

Comment:
If we don't support observed sites, how do we handle a model with a single observed from a compound distribution `pyro.sample('obs', BetaBinomial(c1, c0, count), obs=...)`?

Reply:
Good point, I think we can actually remove this using something like
```py
if obs is None:
    value = pyro.sample("{}_obs".format(name), fn)
else:
    value = obs
```

Reply:
I see, I think we'll also need to convert the `latent` in the `else` block into a deterministic site in that case, and add the log prob contribution from the observed `value`.